src/vector/geometry.rs

```diff
@@ -151,9 +203,42 @@ impl Geometry {
         (x, y, z)
     }
 
-    pub fn get_point_vec(&self) -> Vec<(f64, f64, f64)> {
+    /// Get point coordinates from a line string or a point geometry.
+    ///
+    /// `index` is the line string vertex index, from 0 to `point_count()-1`, or `0` when a point.
+    ///
+    /// See: [`OGR_G_GetPointZM`](https://gdal.org/en/stable/api/vector_c_api.html#_CPPv416OGR_G_GetPointZM12OGRGeometryHiPdPdPdPd)
+    pub fn get_point_zm(&self, index: i32) -> (f64, f64, f64, f64) {
+        let mut x: c_double = 0.;
+        let mut y: c_double = 0.;
+        let mut z: c_double = 0.;
+        let mut m: c_double = 0.;
+        unsafe { gdal_sys::OGR_G_GetPointZM(self.c_geometry(), index, &mut x, &mut y, &mut z, &mut m) };
+        (x, y, z, m)
+    }
+
+    /// Appends all points of a line string to `out_points`. 
+    ///
+    /// Only wkbPoint[X] or wkbLineString[X] may alter `out_points`. Other geometry types will silently do nothing, see
+    /// [`OGR_G_GetPointCount`](https://gdal.org/en/stable/api/vector_c_api.html#_CPPv419OGR_G_GetPointCount12OGRGeometryH)
+    pub fn get_point_vec(&self, out_points: &mut Vec<(f64, f64, f64)>) {
+        // Consider replacing logic with
+        // [OGR_G_GetPoints](https://gdal.org/en/stable/api/vector_c_api.html#_CPPv415OGR_G_GetPoints12OGRGeometryHPviPviPvi)
+        let length = unsafe { gdal_sys::OGR_G_GetPointCount(self.c_geometry()) };
+        out_points.extend(
+            (0..length).map(|i| self.get_point(i))
+        );
+    }
+
+    /// Appends all points of a line string to `out_points`. 
+    ///
+    /// Only wkbPoint[X] or wkbLineString[X] may alter `out_points`. Other geometry types will silently do nothing, see
+    /// [`OGR_G_GetPointCount`](https://gdal.org/en/stable/api/vector_c_api.html#_CPPv419OGR_G_GetPointCount12OGRGeometryH)
+    pub fn get_point_vec_zm(&self, out_points: &mut Vec<(f64, f64, f64, f64)>) {
```

Comment:
I guess these should return an `usize` so it's easier to tell how many points they added.

The changelog needs to be updated too.